opwen_email_client/webapp/actions.py

```diff
@@ -223,3 +229,66 @@ def __call__(self):
         finally:
             if connection is not None:
                 connection.terminate()
+
+
+class ClientRegister(object):
+    def __init__(self, client_name: str, username: str, access_token: str, logger: Logger):
+        self._client_name = client_name
+        self._github_username = username
+        self._github_access_token = access_token
+        self._log = logger or getLogger(__name__)
+
+    @property
+    def client_domain(self):
+        return '{}.{}'.format(self._client_name, 'lokole.ca')
+
+    @property
+    def client_url_create(self):
+        return 'https://{}/api/email/register/'.format('mailserver.lokole.ca')
+
+    @property
+    def client_url_details(self):
+        return 'https://{}/api/email/register/{}'.format('mailserver.lokole.ca', self.client_domain)
+
+    def __call__(self):
+        create_request_payload = dumps({'domain': self.client_domain}).encode('utf-8')
+        create_request = Request(self.client_url_create)
```

Comment:
Given that this code runs in the context of the app where we already installed dependencies, feel free to use [requests.post](https://requests.readthedocs.io/en/master/user/quickstart/#make-a-request). The only reason the install script uses the standard library request object is because it can't have any dependencies as it's what's bootstrapping the entire setup process. There's a tradeoff here of course between keeping the code similar to what's in the installer script, but given that the UX here anyways benefits from moving part of this code into a synchronous action (to provide feedback to the user in case the password is incorrect or some other error happened) and part of it into a background process (to avoid timeout issues waiting for the registration to finish), I suspect the final version of this code will look quite different to the installer anyways.

Reply:
Interesting for sure! requests.post will be a bit nicer to use

Reply:
The plan is to put the create_request as a synchronous task that is part of the form, then moving the stuff in the loop (fetching the details) to celery and putting it in the background?

Reply:
Yes, that's what I'd suggest @adamsclafani.